docker-compose-proxy.yml

```diff
@@ -0,0 +1,26 @@
+version: '3.3'
```

Comment:
I think you're getting it. `docker-compose` cli and `docker-compose.yml` file have different scopes and purposes. The yaml file is used in dev+test+prod (everywhere). It is used by both `docker-compose` cli for dev/test and `docker` cli for prod swarm. Different parts of that file work in different scenarios.

No DAB needed in swarm, DAB is legacy and you can use compose file directly.

### Local dev: 
`docker-compose up`
That uses `docker-compose.yml` to start all the containers with proper settings and build stuff. `build:` objects under each service in compose file are used by `docker-compose` cli only and ignored by `docker stack` commands. [See docs on `build:` in yaml](https://docs.docker.com/compose/compose-file/#build)

### Docker Swarm Prod:
`docker stack deploy -c docker-compose.yml mystackname`
Will deploy all the containers with proper deploy settings like how many replicas of each container to run, healthchecks, how to update the service, etc. [See `deploy:` docs](https://docs.docker.com/compose/compose-file/#deploy)